### ISSUE TITLE
Dynamic change databases on the FLY.

### ISSUE THREAD

**Issue by Nishchit14** [balderdashy/sails-mongo#377]
We are working on multiple databases in single project. Now we have plan to migrate it with sails.
So is there any way to change database as per request.

Scenario looks like.

1) db1.jsbot.io
2) db2.jsbot.io

This two request have their own db scope.

**Comment by devinivy** [comment]
I would suggest creating two different collections that differ only by their configured connection.

**Comment by Nishchit14** [comment]
@devinivy Thanks for quick replay 

But there is possibility of d1,d2,...,dn-1, dn databases.

**Comment by devinivy** [comment]
Are you using them as replicas?  Master/slave?  I don't quite understand the use-case.  If the mongo driver supports what you're trying to do, you should be able to configure sails-mongo to do it.

**Comment by Nishchit14** [comment]
@devinivy 

      The concept is every subdomain has their own database independently. Same as slack concept.

**Comment by devinivy** [comment]
Ah, I see.  Waterline does not support multi-tenancy out of the box.  It might be worth looking at https://github.com/balderdashy/sails-mongo/issues/130 and the stale PR that attempted to implement multi-tenancy in waterline itself https://github.com/balderdashy/waterline/pull/787.  Would you be interested in working on that old PR?  Maybe it could be revived.  A lot of work has already been done.

**Comment by Nishchit14** [comment]
@devinivy 

Thank you for your answer, For temporary we use one temporary solution for change DB.

And i am interested in work with old PR. But i need to check out all the things which is already done and what to do next with it ?

And What is better and efficient way for this , to work in PR or develop new adapter for this ?

**Comment by devinivy** [comment]
I think the best option is to have multi-tenancy supported by waterline, but the most efficient option is to modify an existing adapter to support multi-tenancy.

**Comment by Nishchit14** [comment]
Ok then i will take a look at sails-mongo & waterline.

**Comment by particlebanana** [comment]
Hey @Nishchit14 if you would like to submit a proposal to the [Waterline Roadmap](https://github.com/balderdashy/waterline/blob/master/ROADMAP.md) with a spec of the API you have in mind that would be great! We recently updated our [Contribution Guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) for how we accept new features like this.

Multi-tenancy is definitely something we can achieve once we re-factor the connection handling for transactions.

There is more details on how to do it in this [Stack Overflow](http://stackoverflow.com/questions/29454176/sails-js-i-want-to-add-db-connection-dynamically-after-sails-lift/34845674#34845674) answer.

**Comment by sailsbot** [comment]
Thanks for posting, @Nishchit14.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
